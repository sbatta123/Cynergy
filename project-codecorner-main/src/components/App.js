import React from 'react';
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import '../App.css';
import { NotificationsProvider } from '../contexts/Notifications/NotificationsContext';
import Router from './Router';

function App() {
  return (
    <NotificationsProvider>
      <div className="App">
        <Router />
        <ToastContainer />
      </div>
    </NotificationsProvider>
  );
}

export default App;
