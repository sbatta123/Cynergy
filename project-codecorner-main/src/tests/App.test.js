/**
* @jest-environment jsdom
*/
import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import '@testing-library/jest-dom';
import TestRenderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import App from '../components/App';

// npm start frontend and npm start backend before running these tests

test('renders title', () => {
  render(<App />);
  const linkElement = screen.getByText('Cynergy');
  expect(linkElement).toBeInTheDocument();
});

test('renders login screen login box content', () => {
  render(<App />);
  const loginUsernameInput = screen.getByPlaceholderText(/Enter username/i);
  expect(loginUsernameInput).toBeInTheDocument();
  const loginPasswordInput = screen.getByPlaceholderText(/Enter password/i);
  expect(loginPasswordInput).toBeInTheDocument();
  const submitButton = screen.getByText('Submit');
  expect(submitButton).toBeInTheDocument();
});

test('renders create account', () => {
  render(<App />);
  const linkElement = screen.getByText('Create Account');
  expect(linkElement).toBeInTheDocument();
});

test('renders reset password', () => {
  render(<App />);
  const linkElement = screen.getByText('Reset Password');
  expect(linkElement).toBeInTheDocument();
});

test('user info input', () => {
  render(<App />);
  userEvent.type(screen.getByPlaceholderText(/Enter username/i), 'EmmaONeil');
  expect(screen.getByPlaceholderText(/Enter username/i)).toHaveValue('EmmaONeil');

  userEvent.type(screen.getByPlaceholderText(/Enter password/i), 'cis350project');
  expect(screen.getByPlaceholderText(/Enter password/i)).toHaveValue('cis350project');
});

test('snapshot testing', () => {
  const component = TestRenderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('user clicks submit button with incorrect password inputted', async () => {
  render(
    <App />,
  );
  const user = {
    user: 'EmmaONeil',
    password: 'notpassword',
  };
  const usernameInput = screen.getByPlaceholderText(/Enter username/i);
  const passwordInput = screen.getByPlaceholderText(/Enter password/i);
  const buttonSubmit = screen.getByText('Submit');
  userEvent.type(usernameInput, user.user);
  userEvent.type(passwordInput, user.password);

  userEvent.click(buttonSubmit);
  await waitFor(() => {
    expect(screen.getByRole('alert')).toHaveTextContent('incorrect password provided');
  });
});

test('user clicks submit button with correct password inputted', async () => {
  render(
    <App />,
  );
  const user = {
    user: 'EmmaONeil',
    password: 'hello',
  };
  const usernameInput = screen.getByPlaceholderText(/Enter username/i);
  const passwordInput = screen.getByPlaceholderText(/Enter password/i);
  const buttonSubmit = screen.getByText('Submit');
  userEvent.type(usernameInput, user.user);
  userEvent.type(passwordInput, user.password);

  userEvent.click(buttonSubmit);

  await waitFor(() => {
    const submitButton = screen.queryByText('Submit');
    expect(submitButton).not.toBeInTheDocument();
  });
});
