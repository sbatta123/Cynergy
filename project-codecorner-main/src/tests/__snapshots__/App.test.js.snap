// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <h1
      className="title"
    >
      Cynergy
    </h1>
    <div
      className="score_container"
    >
      <div />
      <img
        alt="cynergy logo"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="login_container card"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="mb-3"
          >
            <input
              className="form-control"
              id="username"
              onChange={[Function]}
              placeholder="Enter username"
            />
          </div>
          <div
            className="mb-3"
          >
            <input
              className="form-control"
              id="password"
              onChange={[Function]}
              placeholder="Enter password"
              type="password"
            />
          </div>
          <button
            className="btn btn-primary"
            disabled={false}
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
    <a
      href="/register"
    >
      Create Account
    </a>
    <a
      href="/resetpassword"
    >
      Reset Password
    </a>
  </div>
  <div
    className="Toastify"
  />
</div>
`;
